src/sock.c

```diff
@@ -99,6 +100,17 @@ sock_t sock_connect(const char * const host, const unsigned int port)
         if (sock < 0)
             continue;
 
+	/* turn on keepalive */
```

Comment:
I see. Just my thoughts how it can be implemented:
 * Optional feature that is enabled with passing `XMPP_CONN_FLAG_KEEPALIVE` to `xmpp_conn_set_flags()`. This is more preferable than adding new interface `xmpp_conn_set_keepalive()`.
 * Enabled by default, like you did in your patch. This would change default behaviour of libstrophe. Also there are situations when network devices switch to a powersave mode (for example, mobile platforms), sending keepalives by default makes the devices be active without options.

Anyway, enabling keepalives can lead to some unpredictable behaviour or errors, so I would defer this change until I make some tests.

One more option that would help you to keep connection alive is xmmp ping (see http://xmpp.org/extensions/xep-0199.html). An example: https://github.com/boothj5/profanity/blob/master/src/xmpp/stanza.c#L1025